File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/工厂模式/java/ComputerFactory.java
package com.redkey.kotlintest.kotlin.design_pattern.工厂模式.java;

public class ComputerFactory {
    public static final String FACTORY_PC = "pc";
    public static final String FACTORY_SERVER = "server";

    public static Computer getComputer(String type, String ram, String hdd, String cpu) {
        if ("PC".equalsIgnoreCase(type)) return new PC(ram, hdd, cpu);
        else if ("Server".equalsIgnoreCase(type)) return new Server(ram, hdd, cpu);

        return null;
    }
}

File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/工厂模式/kotlin/ComputerFactory.kt
package com.redkey.kotlintest.kotlin.design_pattern.工厂模式.kotlin

class ComputerFactory {
    companion object {
        const val FACTORY_PC: String = "pc"
        const val FACTORY_SERVER: String = "server"

        fun getComputer(type: String, ram: String, hdd: String, cpu: String): Computer? {
            return when {
                type.equals(FACTORY_PC, false) -> PC(ram, hdd, cpu)
                type.equals(FACTORY_SERVER, false) -> Server(ram, hdd, cpu)
                else -> null
            }
        }
    }

}